lib/Sender.js

```diff
@@ -41,13 +41,19 @@ class Sender extends EventEmitter {
     dataBuffer.writeUInt16BE(code, 0);
     if (dataBuffer.length > 2) dataBuffer.write(data, 2);
 
-    var self = this;
-    this.messageHandlers.push(function (callback) {
-      self.frameAndSend(0x8, dataBuffer, true, mask);
-      callback();
-      if (typeof cb === 'function') cb();
-    });
-    this.flush();
+    if (this.extensions[PerMessageDeflate.extensionName]) {
+      this.enqueue([this.doClose, [dataBuffer, mask, cb]]);
+    } else {
+      this.doClose(dataBuffer, mask, cb);
+    }
+  }
+
+  doClose (data, mask, cb) {
+    this.frameAndSend(0x8, data, true, mask);
+    if (this.extensions[PerMessageDeflate.extensionName]) {
+      this.messageHandlerCallback();
+    }
+    if (typeof cb === 'function') cb();
```

Comment:
I think we can avoid checking the type of the callback, here and throughout the file.